cmd/kubeadm/app/constants/constants_test.go

```diff
@@ -149,50 +148,62 @@ func TestAddSelfHostedPrefix(t *testing.T) {
 }
 
 func TestEtcdSupportedVersion(t *testing.T) {
+	var supportedEtcdVersion = map[uint8]string{
+		13: "3.2.24",
+		14: "3.3.10",
+		15: "3.3.10",
+		16: "3.3.17-0",
+		17: "3.4.3-0",
+		18: "3.4.3-0",
+	}
 	var tests = []struct {
 		kubernetesVersion string
 		expectedVersion   *version.Version
+		expectedWarning   error
 		expectedError     error
```

Comment:
i think we should just have `bool for `expectedWarning` and `expectedError`  here.
no need to verify the error types.

e.g.
```
if (actualWarning != nil) != rt.expectedWarning {
...
}
```